File: app/src/main/java/co/edu/unipiloto/myapplication/model/ShippingStatus.java
package co.edu.unipiloto.myapplication.model;

import com.google.gson.annotations.SerializedName;

public class ShippingStatus {
    // Mapea los campos reales de la API (JSONPlaceholder) a tus nombres lógicos:
    @SerializedName("id")
    private String id; // El ID de la guía

    @SerializedName("title")
    private String status; // Usamos 'title' del JSON como el 'status'

    @SerializedName("body")
    private String destinationAddress; // Usamos 'body' del JSON como la 'dirección'

    // Estos campos no existen en JSONPlaceholder, los dejamos null/vacío para la prueba.
    private String estimatedDate = "2025-12-31";
    private String timeFranja = "PM";

    // Getters necesarios:
    public String getId() { return id; }
    public String getStatus() { return status; }
    public String getDestinationAddress() { return destinationAddress; }
    public String getEstimatedDate() { return estimatedDate; }
    public String getTimeFranja() { return timeFranja; }
}

File: app/src/main/java/co/edu/unipiloto/myapplication/model/ShippingStatus.kt
package co.edu.unipiloto.myapplication.model

/**
 * Modelo de datos para mostrar el estado de seguimiento (Tracking) en la UI.
 * Este modelo refleja los datos que DEBE retornar el endpoint de Tracking del backend.
 */
data class ShippingStatus(
    // 1. Identificador de Guía (Usaremos el tracking number real del backend)
    // El backend podría devolver el número de guía completo.
    val trackingNumber: String,

    // 2. Estado (PENDIENTE, EN RUTA, ENTREGADO)
    val status: String,

    // 3. Dirección de Destino Completa
    val destinationAddress: String,

    // 4. Fecha y Franja de Entrega (Campos generados por tu antigua lógica, útiles para la UI)
    val estimatedDate: String,
    val timeFranja: String
) {
}